src/azure-cli-core/azure/cli/core/cloud.py

```diff
@@ -238,23 +238,12 @@ def get_custom_clouds():
     return [c for c in get_clouds() if c.name not in known_cloud_names]
 
 
-def init_known_clouds(force=False):
-    config = get_config_parser()
-    config.read(CLOUD_CONFIG_FILE)
-    stored_cloud_names = config.sections()
-    for c in KNOWN_CLOUDS:
-        if force or c.name not in stored_cloud_names:
-            _config_add_cloud(config, c, overwrite=force)
-    if not os.path.isdir(GLOBAL_CONFIG_DIR):
-        os.makedirs(GLOBAL_CONFIG_DIR)
-    with open(CLOUD_CONFIG_FILE, 'w') as configfile:
-        config.write(configfile)
-
-
 def get_clouds():
     clouds = []
-    # load the config again as it may have changed
     config = get_config_parser()
+    # Start off with known clouds and apply config file on top of current config
```

Comment:
A more detailed description of the logic of this method would be beneficial to all the future code reader.
1. The known clouds will be always added to the list;
2. The clouds definition in the profile file will be added to the list;
3. The known clouds can be rewritten in the profile.